clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java

```diff
@@ -258,11 +297,72 @@ private VoltronReadWriteLock createAccessLockFor(String entityIdentifier) {
 
   private EntityRef<EhcacheClientEntity, ClusteredTierManagerConfiguration> getEntityRef(String identifier) {
     try {
-      return connection.getEntityRef(EhcacheClientEntity.class, EhcacheEntityVersion.ENTITY_VERSION, identifier);
+      return connection.getEntityRef(EhcacheClientEntity.class, ENTITY_VERSION, identifier);
     } catch (EntityNotProvidedException e) {
       LOGGER.error("Unable to get clustered tier manager for id {}", identifier, e);
       throw new AssertionError(e);
     }
   }
 
+  public ClusteredTierClientEntity fetchOrCreateClusteredStoreEntity(UUID clientId, String clusterTierManagerIdentifier,
+                                                                     String storeIdentifier, ServerStoreConfiguration clientStoreConfiguration,
+                                                                     boolean autoCreate) throws EntityNotFoundException, CachePersistenceException {
+    EntityRef<ClusteredTierClientEntity, ClusteredTierEntityConfiguration> entityRef;
+    try {
+      entityRef = connection.getEntityRef(ClusteredTierClientEntity.class, ENTITY_VERSION, clusterTierManagerIdentifier + storeIdentifier);
```

Comment:
Why are we not using the same acquireleadership/lock concept for cache-entity lifecycle as it is used in cachemanager-entity lifecycle. IMHO the purpose served by acquireleadership/lock concept should apply to both caches and cachemanager as far as their respective entities are concerned. That way in ehcache we can have only one way. Just that in case of cache the lock will for that cache, wdyt ?